### PR TITLE
feature(update): allow updating apps from local manifest or URL

### LINKED ISSUE #3607: Update of manifests installed by url fails
Reproducible steps:

1. `scoop bucket rm Ash258` in case you already have this bucket added.
1. `scoop install 'https://raw.githubusercontent.com/Ash258/scoop-Ash258/master/bucket/RPCS3.json'`
    - RPCS3 is good example as it has frequent updates
1. Wait for update of manifest
1. `scoop update`
1. `scoop status`
    - Status should find out there was update of manifest
1. `scoop update RPCS3`
1. `Could not find manifest for RPCS3`

Same problem with Rider manifest

image

For debug purpose I suggest to create custom gist and just change version property.

### LINKED ISSUE #3683: Issue when using scoop update from a manifest url
**What:**
We can install an application from a local manifest (url), we should be able to update it from the same url

**Use case:**
1. Create an application manifest on your machine `C:/scoop_manifests/appA.json`
2. install an app from the local manifest on your machine:
`scoop install C:/scoop_manifests/appA.json`
3. Update the local manifest `C:/scoop_manifests/appA.json`
4. Update the app with scoop
`scoop update appA`

**Expected:**
This should update the app with your new manifest value

**Actual result:**
Scoop look in its buckets if the app exist, if so it update the app to its latest bucket version

**Investigation:**
in scoop-update.ps1 we don't concider if the manifest contains an url instead of an bucket name

### PR DESCRIPTION
**Pull request by stephanec1**
- Closes #3683 
- Closes #3607

### CODE REVIEW

**Comment by Ash258** [comment]
This is not enough. You need to cover dependcy resolver too.

Update will fail on function `update` on `$(deps $app $architecture)`

**Comment by stephanec1** [comment]
moreover its not an elseif but an else, we always set the bucket name to "main" if it doesn't exist

**Comment by stephanec1** [comment]
Dependencies do not support url manifest for updates. 
The code just check if there is an app with the same name than the dependency. But if the dependency is a manifest.json, if didn't find anything

File: libexec/scoop-update.ps1 [hunk 1]
```diff
@@ -173,6 +173,11 @@ function update($app, $global, $quiet = $false, $independent, $suggested, $use_c
 
     if (!$independent) {
         # check dependencies
+        if ($install.url) {
+            $deps = @(deps $install.url $architecture) | Where-Object { !(installed $_) }
+        } else {
+            $deps = @(deps $app $architecture) | Where-Object { !(installed $_) }
+        }
         $deps = @(deps $app $architecture) | Where-Object { !(installed $_) }
```
Comment (Ash258) [inline libexec/scoop-update.ps1]:
```suggestion
		$man = if ($url) { $url } else { $app }
        $deps = @(deps $man $architecture) | Where-Object { !(installed $_) }
```

File: libexec/scoop-update.ps1 [hunk 2]
```diff
@@ -173,7 +173,13 @@ function update($app, $global, $quiet = $false, $independent, $suggested, $use_c
 
     if (!$independent) {
         # check dependencies
-        $deps = @(deps $app $architecture) | Where-Object { !(installed $_) }
+        if ($install.url) {
+            $deps = @(deps $install.url $architecture) | Where-Object { !(installed $_) }
+        } else {
+            $deps = @(deps $app $architecture) | Where-Object { !(installed $_) }
+        }
```
Comment (Ash258) [inline libexec/scoop-update.ps1]: Remove these obsolete lines.

Comment (stephanec1) [reply libexec/scoop-update.ps1]: oups sorry

**Comment by Ash258** [comment]
Add snippet bellow as PR description

```txt
- Closes #3683 
- Closes #3607
```

**Pull request merged**